lib/iterator.js

```diff
@@ -199,6 +199,68 @@ class Iterator {
     return result + suffix;
   }
 
+  // Find value in this iterator compared by `comparator` on values
```

Comment:
Maybe clarify the way it works: 'Find value in this iterator by comparing every value with the found one using `comparator`'